src/Compiler/pars.fsy

```diff
@@ -6268,11 +6268,13 @@ measureTypePower:
   | measureTypeAtom INFIX_AT_HAT_OP atomicRationalConstant
      { if $2 <> "^" && $2 <> "^-" then reportParseErrorAt (rhs parseState 2) (FSComp.SR.parsUnexpectedOperatorForUnitOfMeasure())
        if $2 = "^-" then
-           let afterMinus = (rhs parseState 2).EndRange
+           let mOp = rhs parseState 2
+           let afterMinus = mOp.EndRange
            let beforeMinus = mkRange afterMinus.FileName (mkPos afterMinus.EndLine (afterMinus.EndColumn - 1)) afterMinus.End
-           let m = unionRanges beforeMinus (rhs parseState 3)  // include MINUS in Negate range
-           SynMeasure.Power($1, SynRationalConst.Negate($3, m), lhs parseState)
-       else SynMeasure.Power($1, $3, lhs parseState) }
+           let mNegate = unionRanges beforeMinus (rhs parseState 3)  // include MINUS in Negate range
+           let mCaret = unionRanges mOp.StartRange mNegate.StartRange
```

Comment:
The change looks corrects, but it seems that things get much more complicated than they should here. I think there're a few other places where ranges are created this way across the compiler and other tooling.

What if we had APIs to extend/shrink ranges in `Range` module instead? From the top of my head:

```fsharp
val withStart : startPos: pos -> m: range -> range
val withEnd : endPos: pos -> m: range -> range
```
or
```fsharp
val shiftStart : lineDelta: int -> columnDelta: int -> m: range -> range
val shiftEnd : lineDelta: int -> columnDelta: int -> m: range -> range

// or even more granular
val shiftEndColumn : delta: int -> m: range -> range
...
```

It would not require users to deal with file names (and looking up these names in range index table), and constructing the positions for the range, making it something like this here:

```fsharp
let mMinus = shiftEndColumn -1 mOp
```

Reply:
Yep, I think that would be useful. :)